packages/invoices/src/InboundShipment/DetailView/DetailView.tsx

```diff
@@ -1,60 +1,108 @@
 import React, { FC } from 'react';
 import { useParams } from 'react-router';
 import {
-  Column,
+  useQuery,
   TableProvider,
   createTableStore,
   useOmSupplyApi,
-  useDocument,
   useColumns,
   GenericColumnKey,
   getNotePopoverColumn,
   getRowExpandColumn,
   useDialog,
   DialogButton,
   useTranslation,
-  Item,
+  useMutation,
+  useQueryClient,
 } from '@openmsupply-client/common';
 
-import { InboundAction, reducer } from './reducer';
 import { getInboundShipmentDetailViewApi } from './api';
 import { Toolbar } from './Toolbar';
 import { Footer } from './Footer';
 import { AppBarButtons } from './AppBarButtons';
 import { SidePanel } from './SidePanel';
 import { GeneralTab } from './GeneralTab';
 import { InboundLineEdit } from './modals/InboundLineEdit/InboundLineEdit';
-
-import { isInboundEditable } from '../../utils';
-import { InboundShipmentItem } from '../../types';
+import { inboundLinesToSummaryItems } from './reducer/reducer';
+import {
+  getNextInboundStatus,
+  isInboundEditable,
+  placeholderInbound,
+} from '../../utils';
+import {
+  InboundShipmentItem,
+  InboundShipment,
+  Invoice,
+  OutboundShipmentSummaryItem,
+} from '../../types';
 
 const useDraftInbound = () => {
+  const queryClient = useQueryClient();
   const { id } = useParams();
   const { api } = useOmSupplyApi();
+  const queries = getInboundShipmentDetailViewApi(api);
 
-  const { draft, save, dispatch, state } = useDocument(
-    ['invoice', id],
-    reducer,
-    getInboundShipmentDetailViewApi(api)
-  );
+  const onChangeSortBy = () => {};
+
+  const { data } = useQuery(['invoice', id], () => {
+    return queries.onRead(id);
+  });
+
+  const draft = data
+    ? { ...data, items: inboundLinesToSummaryItems(data?.lines ?? []) }
+    : placeholderInbound;
+
+  const { mutateAsync } = useMutation(queries.onUpdate, {
+    onMutate: async (patch: Partial<InboundShipment>) => {
+      await queryClient.cancelQueries(['invoice', id]);
+
+      const previousInbound: Invoice = queryClient.getQueryData([
+        'invoice',
+        id,
+      ]);
+
+      queryClient.setQueryData(['invoice', id], {
+        ...previousInbound,
+        ...patch,
+      });
+
+      return { previousInbound, patch };
+    },
+    onSettled: () => queryClient.invalidateQueries(['invoice', id]),
+    onError: (_, __, context) => {
+      queryClient.setQueryData(['invoice', id], context.previousInbound);
+    },
+  });
+
+  const { isLoading: isAddingItem, mutateAsync: noOptimisticMutate } =
```

Comment:
should that be:
```suggestion
  const { isLoading: isAddingItem, mutateAsync: nonOptimisticMutate } =
```
wasn't sure.